src/wrappers/themis/rust/src/secure_session.rs

```diff
@@ -863,18 +882,22 @@ unsafe extern "C" fn get_public_key_for_id(
     key_len: usize,
     user_data: *mut c_void,
 ) -> c_int {
-    let id = byte_slice_from_ptr(id_ptr as *const u8, id_len);
-    let key_out = byte_slice_from_ptr_mut(key_ptr as *mut u8, key_len);
-    let transport = &mut user_data_as_context(user_data).transport;
-
-    if let Some(key) = transport.get_public_key_for_id(id) {
-        let key = key.as_ref();
-        if key_out.len() >= key.len() {
-            key_out[0..key.len()].copy_from_slice(key);
-            return 0;
+    let result = panic::catch_unwind(|| {
+        let id = byte_slice_from_ptr(id_ptr as *const u8, id_len);
+        let key_out = byte_slice_from_ptr_mut(key_ptr as *mut u8, key_len);
+        let transport = &mut user_data_as_context(user_data).transport;
+
+        if let Some(key) = transport.get_public_key_for_id(id) {
+            let key = key.as_ref();
+            if key_out.len() >= key.len() {
+                key_out[0..key.len()].copy_from_slice(key);
+                return 0;
+            }
         }
-    }
-    -1
+
+        -1
```

Comment:
what mean `-1`? it's correct expression?

Reply:
It means "return `-1` from this function".

In Rust the last expression in the function is its return value. I personally prefer to spell it out explicitly with `return` if it's not obvious or if early returns are user nearby. However, automatic code style checkers (rustfmt and clippy) don't like that form and enforce the implicit one.

Oh, and looking at it now I guess I'll give names to magic constants here as well.